engine/access/rest/blocks.go

```diff
@@ -0,0 +1,250 @@
+	if err != nil {
+		return nil, NewBadRequestError(err)
+	}
+
+	blocks := make([]*generated.Block, len(ids))
+	for i, id := range ids {
+		block, err := getBlock(forID(&id), r, backend, link)
+		if err != nil {
+			return nil, err
+		}
+		blocks[i] = block
+	}
+
+	return blocks, nil
+}
+
+func getBlocksByHeight(r *request, backend access.API, link LinkGenerator) (interface{}, error) {
+	heights := r.getQueryParams(heightQueryParam)
+	startHeight := r.getQueryParam(startHeightQueryParam)
+	endHeight := r.getQueryParam(endHeightQueryParam)
+
+	// if both height and one or both of start and end height are provided
+	if len(heights) > 0 && (startHeight != "" || endHeight != "") {
+		err := fmt.Errorf("can only provide either heights or start and end height range")
+		return nil, NewBadRequestError(err)
+	}
+
+	// if neither height nor start and end height are provided
+	if len(heights) == 0 && (startHeight == "" || endHeight == "") {
+		err := fmt.Errorf("must provide either heights or start and end height range")
+		return nil, NewBadRequestError(err)
+	}
+
+	// if the query is /blocks?height=final or /blocks?height=sealed, lookup the last finalized or the last sealed block
+	if len(heights) == 1 && (heights[0] == finalHeightQueryParam || heights[0] == sealedHeightQueryParam) {
+		block, err := getBlock(forFinalized(heights[0]), r, backend, link)
+		if err != nil {
+			return nil, err
+		}
+
+		return []*generated.Block{block}, nil
+	}
+
+	// if the query is /blocks/height=1000,1008,1049...
+	if len(heights) > 0 {
+		uintHeights, err := toHeights(heights)
+		if err != nil {
+			heightError := fmt.Errorf("invalid height specified: %v", err)
+			return nil, NewBadRequestError(heightError)
+		}
+
+		blocks := make([]*generated.Block, len(uintHeights))
+		for i, h := range uintHeights {
+			block, err := getBlock(forHeight(h), r, backend, link)
+			if err != nil {
+				return nil, err
+			}
+			blocks[i] = block
+		}
+
+		return blocks, nil
+	}
+
+	// support providing end height as "sealed" or "final"
+	if endHeight == finalHeightQueryParam || endHeight == sealedHeightQueryParam {
+		latest, err := backend.GetLatestBlock(r.Context(), endHeight == sealedHeightQueryParam)
+		if err != nil {
+			return nil, err
+		}
+
+		endHeight = fmt.Sprintf("%d", latest.Header.Height)
+	}
+
+	// lookup block by start and end height range
+	start, err := toHeight(startHeight)
+	if err != nil {
+		heightError := fmt.Errorf("invalid start height %s: %v", startHeight, err)
+		return nil, NewBadRequestError(heightError)
+	}
+	end, err := toHeight(endHeight)
+	if err != nil {
+		heightError := fmt.Errorf("invalid end height %s: %v", endHeight, err)
+		return nil, NewBadRequestError(heightError)
+	}
+
+	if start > end {
+		err := fmt.Errorf("start height must be less than or equal to end height")
+		return nil, NewBadRequestError(err)
+	}
+
+	if end-start > MaxAllowedHeights {
+		err := fmt.Errorf("height interval exceeding maximum interval of %d", MaxAllowedHeights)
```

Comment:
this wording is a little confusing.
```suggestion
		err := fmt.Errorf("height range %d exceeds maximum allowed of %d", end-start, MaxAllowedHeights)
```